source/dofs/dof_handler.cc

```diff
@@ -2602,6 +2602,16 @@ DoFHandler<dim, spacedim>::distribute_dofs(
   // hand the actual work over to the policy
   this->number_cache = this->policy->distribute_dofs();
 
+  // cache locally-relevant degrees of freedom in the form of a partitioner
+  {
+    IndexSet locally_relevant_dofs;
+    DoFTools::extract_locally_relevant_dofs(*this, locally_relevant_dofs);
```

Comment:
I think that we should move the code that computes the locally relevant dofs here and in `DoFTools` simply return the the union `partitioner->locally_owned_range()` and `partitioner->ghost_indices()`. The idea is that we call this function in a number of places, and now that we have the information around we should use it to simplify those tasks.

Reply:
I like this suggestion. I'll implement this tomorrow!